py3status/modules/year_progress.py

```diff
@@ -0,0 +1,212 @@
+    progress_block = '▓'
+    remain_block = '░'
+    progress_width = 5
+    ctime = ["2020-12-16 20:59:00", "%Y-%m-%d %H:%M:%S"]
+}
+```
+
+@author azzamsa <https://github.com/azzamsa>
+@license GPLv3 <https://www.gnu.org/licenses/gpl-3.0.txt>
+
+Notes:
+    Credit: Part of the code are ported from twitter.com/year_progress <https://github.com/filiph/progress_bar>
+
+SAMPLE OUTPUT
+{'full_text': '▓▓▓▓░ 87%y'}
+
+{'full_text': '▓▓▓░░ 50%m'}
+
+{'full_text': '▓▓░░░ 33%d'}
+
+{'full_text': '▓▓▓▓░ 83%w'}
+"""
+
+from datetime import datetime, tzinfo, timedelta
+
+
+class UTC(tzinfo):
+    """UTC"""
+
+    def utcoffset(self, dt):
+        return timedelta(0)
+
+    def tzname(self, dt):
+        return "UTC"
+
+    def dst(self, dt):
+        return timedelta(0)
+
+
+def compute_progress(start, end, current):
+    whole_diff = end - start
+    whole_diff_in_seconds = whole_diff.days * 86400 + whole_diff.seconds
+
+    if whole_diff_in_seconds == 0:
+        raise ValueError("Start and end datetimes are equal.")
+    current_diff = current - start
+    current_diff_in_seconds = current_diff.days * 86400 + current_diff.seconds
+
+    return float(current_diff_in_seconds) / float(whole_diff_in_seconds)
+
+
+def compute_current_week_progress(current=None, week_end=None):
+    """Compute current day progress to the end of the week"""
+    if not current:
+        current = datetime.now(tz=UTC())
+
+    week_start = current - timedelta(days=current.weekday())
+    if not week_end:
+        week_end = week_start + timedelta(days=6)
+    return compute_progress(week_start, week_end, current)
+
+
+def compute_current_day_progress(current=None, end=None):
+    """Compute current hour progress to tomorrow (00:00 AM)"""
+    if not current:
+        current = datetime.now().astimezone()
+
+    # start at 00:00 AM
+    start = datetime(current.year, current.month, current.day).astimezone()
+    if not end:
+        end = datetime(
+            current.year, current.month, current.day, 23, 59, 00
+        ).astimezone()
+
+    return compute_progress(start, end, current)
+
+
+def compute_current_month_progress(current=None, end=None):
+    """Compute current date to the first date of  next month"""
+    if not current:
+        current = datetime.now(tz=UTC())
+
+    start = datetime(current.year, current.month, 1, tzinfo=UTC())
+    if not end:
+        # month max 12
+        if current.month == 12:
+            end = datetime(current.year, current.month, 31, tzinfo=UTC())
+        else:
+            end = datetime(current.year, current.month + 1, 1, tzinfo=UTC())
+    return compute_progress(start, end, current)
+
+
+def compute_current_year_progress(current=None, end=None):
+    """Compute current year progress to next year"""
+    if not current:
+        current = datetime.now(tz=UTC())
+
+    start = datetime(current.year, 1, 1, tzinfo=UTC())
+    if not end:
+        end = datetime(current.year + 1, 1, 1, tzinfo=UTC())
```

Comment:
(Trivial) It seems like you could do `utc = UTC()` and pass it to those without instantiating more.